htdocs/core/class/html.formcontract.class.php

```diff
@@ -187,8 +187,9 @@ public function formSelectContract($page, $socid = -1, $selected = '', $htmlname
 		print '<form method="post" action="'.$page.'">';
 		print '<input type="hidden" name="action" value="setcontract">';
 		print '<input type="hidden" name="token" value="'.newToken().'">';
-		$this->select_contract($socid, $selected, $htmlname, $maxlength, $showempty, $showRef);
 		print '<input type="submit" class="button smallpaddingimp valignmiddle" value="'.$langs->trans("Modify").'">';
 		print '</form>';
+		return $this->select_contract($socid, $selected, $htmlname, $maxlength, $showempty, $showRef);
```

Comment:
The method formSelectContract make print. So the goal is to output HTML code.
The select_contract() also output HTML code but if you move it inside the return, it means the code will be output out of the `<form> </form>`
Call of select_contract must be restored as its original place and result saved into a $result variable to be output into the return.